usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateCalendarEndDateBeforeStartDate.java

```diff
@@ -51,10 +50,14 @@ public ValidateCalendarEndDateBeforeStartDate(final GtfsDataRepository dataRepo,
      */
     public void execute() {
         logger.info("Validating rule 'E032 - calendar.txt end_date is before start_date'" + System.lineSeparator());
-        Collection<Calendar> calendars = dataRepo.getCalendarAll();
-        calendars.stream()
-                .filter(calendar -> calendar.getEndDate().isBefore(calendar.getStartDate()))
-                .forEach(calendar -> resultRepo.addNotice(new CalendarEndDateBeforeStartDateNotice("calendar.txt",
-                        calendar.getServiceId(), calendar.getStartDate(), calendar.getEndDate())));
+
+        dataRepo.getCalendarAll().forEach((serviceId, calendar) -> {
```

Comment:
Is there a specific reason we dropped the streaming API? wouldn't that be more memory efficient? (I do not know the answer to that question, just thought relevant to raise it)

Reply:
Discussing this on our slack channel, will report the final decision here.

Reply:
Will use the streaming API. It has been dropped just to manipulate the original content returned by method `getCalendarAll`, but it makes more sense to use the streaming model for code legibility and memory performance.